src/r4300/new_dynarec/new_dynarec.c

```diff
@@ -10127,91 +10127,150 @@ int new_recompile_block(int addr)
   // to use, which can avoid a load-use penalty on certain CPUs.
   for(i=0;i<slen-1;i++)
   {
-    if(!i||(itype[i-1]!=UJUMP&&itype[i-1]!=CJUMP&&itype[i-1]!=SJUMP&&itype[i-1]!=RJUMP&&itype[i-1]!=FJUMP))
+    if(count_free_regs(regs[i].regmap)>minimum_free_regs[i])
```

Comment:
I'm not a big fan of nested if statement in general. It make the code harder to read/maintain. I like the "filter approach". The idea is to allow "if not,continue" at the top of a for loop (and only here). This technical "filter" the iterated content before the code really begin. So if you have a if statement taking the whole function block without any else statement it's a good candidate for "filtering".

Roughly:
```c
for loop

 # filter code block (find what could "skip" the loop)

 # loop execution code block
```
Do the separation at the beginning of the for loop make the loop execution code easier to read.